crates/yew_router_macro/src/switch/attribute.rs

```diff
@@ -9,45 +9,50 @@ pub enum AttrToken {
 }
 
 impl AttrToken {
-    pub fn convert_attributes_to_tokens(attributes: Vec<Attribute>) -> Vec<Self> {
-        fn get_meta_name_value_str(mnv: &MetaNameValue) -> Option<String> {
+    pub fn convert_attributes_to_tokens(attributes: Vec<Attribute>) -> syn::Result<Vec<Self>> {
+        fn get_meta_name_value_str(mnv: &MetaNameValue) -> syn::Result<String> {
             match &mnv.lit {
-                Lit::Str(s) => Some(s.value()),
-                _ => None,
+                Lit::Str(s) => Ok(s.value()),
+                lit => Err(syn::Error::new_spanned(lit, "expected a string literal")),
             }
         }
 
         attributes
             .iter()
             .filter_map(|attr: &Attribute| attr.parse_meta().ok())
-            .filter_map(|meta: Meta| match meta {
-                Meta::NameValue(mnv) => mnv
-                    .path
-                    .clone()
-                    .get_ident()
-                    .into_iter()
-                    .filter_map(|ident| match ident.to_string().as_str() {
-                        "to" => Some(AttrToken::To(
-                            get_meta_name_value_str(&mnv)
-                                .expect("Value provided after `to` must be a String"),
-                        )),
-                        "rest" => Some(AttrToken::Rest(Some(
-                            get_meta_name_value_str(&mnv)
-                                .expect("Value provided after `rest` must be a String"),
-                        ))),
-                        _ => None,
-                    })
-                    .next(),
-                Meta::Path(path) => path
-                    .get_ident()
-                    .into_iter()
-                    .filter_map(|ident| match ident.to_string().as_str() {
-                        "end" => Some(AttrToken::End),
-                        "rest" => Some(AttrToken::Rest(None)),
+            .filter_map(|meta: Meta| {
+                let meta_span = meta.span();
+                match meta {
+                    Meta::NameValue(mnv) => mnv.path.get_ident().into_iter().find_map(|ident| {
```

Comment:
All three `.get_ident().into_iter().find_map(...)`s could be further simplified to just `.get_ident().map(...)`. I missed that before but am happy to implement that and maybe more code / error handling improvements in the next PR.

Reply:
Since the PR hasn't been merged yet, I've addressed this now.